packages/cicero-core/src/template.js

```diff
@@ -172,15 +178,60 @@ class Template {
         return hasher.digest('hex');
     }
 
+    /**
+     * signs a string made up of template hash and time stamp using private key derived
+     * from the keystore
+     * @param {String} [p12File] - encoded string of p12 keystore file
+     * @param {String} [passphrase] - passphrase for the keystore file
+     * @param {Number} [timeStamp] - timeStamp of the moment of signature is done
+     * @private
+     */
+    signTemplate(p12File, passphrase, timeStamp) {
+        const templateHash = this.getHash();
+        // decode p12 from base64
+        const p12Der = forge.util.decode64(p12File);
+        // get p12 as ASN.1 object
+        const p12Asn1 = forge.asn1.fromDer(p12Der);
+        // decrypt p12 using the passphrase 'password'
+        const p12 = forge.pkcs12.pkcs12FromAsn1(p12Asn1, false, passphrase);
+        //X509 cert forge type
+        const certificateForge = p12.safeContents[0].safeBags[0].cert;
+        //Private Key forge type
+        const privateKeyForge = p12.safeContents[1].safeBags[0].key;
+        //convert cert and private key from forge to PEM
+        const certificatePem = forge.pki.certificateToPem(certificateForge);
+        const privateKeyPem = forge.pki.privateKeyToPem(privateKeyForge);
+        //convert private key in pem to private key type in node
+        const privateKey = crypto.createPrivateKey(privateKeyPem);
+        const sign = crypto.createSign('SHA256');
+        sign.write(templateHash + timeStamp);
+        sign.end();
+        const signature = sign.sign(privateKey, 'hex');
+        const signatureObject = {
+            templateHash: templateHash,
+            timeStamp: timeStamp,
```

Comment:
Please consider `timestamp` instead here. Also consider using object initialization shorthand notation., e.g.
```
{  templateHash, timestamp, ... }
```

Reply:
Would it be worth writing `$timestamp` instead of `timeStamp` as Concerto does now? More generally should the signature object have a valid CTO model?

Reply:
What will `$` signify in this case?